File: Selenium/Project/activity8.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.testng.annotations.AfterClass;
import org.testng.annotations.AfterMethod;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import java.util.List;
import java.util.concurrent.TimeUnit;

public class activity8 {

    // Declare the WebDriver object
    WebDriver driver;

    @BeforeMethod
    public void beforeMethod() {
        // Set up the Firefox driver
        WebDriverManager.firefoxdriver().setup();
        driver = new FirefoxDriver();

        //Open browser
        driver.get("https://alchemy.hguy.co/crm/");
        driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
    }

    @Test
    public void menuChecking() throws InterruptedException {
        //Find the username and password fields
        WebElement username = driver.findElement(By.id("user_name"));
        WebElement password = driver.findElement(By.id("username_password"));

        //Enter credentials
        username.sendKeys("admin");
        password.sendKeys("pa$$w0rd");

        //Click login
        driver.findElement(By.id("bigbutton")).click();
        //wait
        driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);

        //Initialize Actions Object
        Actions builder = new Actions(driver);

        // navigate to sales->accounts

        WebElement sales = driver.findElement(By.id("grouptab_0"));

        builder.moveToElement(sales).build().perform();


        WebElement accounts = driver.findElement(By.id("moduleTab_9_Accounts"));
        accounts.click();

        Thread.sleep(3000);
        //wait
        driver.manage().timeouts().implicitlyWait(25, TimeUnit.SECONDS);

        List<WebElement> elements = driver.findElements(By.xpath("//table[contains(@class,'responsive')]/tbody/tr[position() mod 2 =1][position()<6]/td[3]"));

        System.out.println("The size of the list elements is: " + elements.size());

        for (WebElement ele : elements) {
            String name = ele.getText();
            System.out.println("The Names of the first 5 Odd Rows are: " + name);
        }




    }

    @AfterMethod
    public void afterMethod() {
        //Close the browser
        driver.quit();
    }

}

File: Selenium/Project/activity8.py
from selenium import webdriver
from selenium.webdriver.common.by import By

from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.ui import WebDriverWait
import time

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())
#locator = Locator()

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Navigate to the URL
    driver.get("http://alchemy.hguy.co/crm")

    # Print the title of the page
    print("Page title is: ", driver.title)
    # Find the username field
    username = driver.find_element(By.ID, "user_name")
    # Find the password field
    password = driver.find_element(By.ID, "username_password")

    # Enter the given credentials
    # Enter username
    username.send_keys("admin")
    # Enter password
    password.send_keys("pa$$w0rd")

    # Find the login button
    login = driver.find_element(By.ID, "bigbutton")
    login.click()
    # Print the login message
    message = driver.find_element(By.CLASS_NAME, "hidden-xs")
    print("Login message: ", message.text)
    # Navigate to Sales.

    Sales = WebDriverWait(driver,10).until(EC.visibility_of_element_located((By.ID,"grouptab_0")))
    Sales.click()

    # Find and click the Sales >> Account submenu
    Account_submenu = WebDriverWait(driver, 10).until(EC.visibility_of_element_located((By.ID, "moduleTab_9_Accounts")))
    Account_submenu.click()
    # Print the Account Page message
    time.sleep(5)
    #total odd data
    oddRows = driver.find_elements(By.XPATH,("//*[@class='oddListRowS1']"))
    print("Total odd data rows found in table:" ,len(oddRows))

    #loop
    for i in range(0, 5):
        row = oddRows[i]

        name = row.find_element(By.XPATH, ".//td[3]").text  # Assuming the name is in the third column (td[3])
        print("Name:", name)
    #close browser
    driver.quit()